cloud/services/computes/droplets.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package computes
+
+import (
+	"encoding/base64"
+	"net/http"
+	"strconv"
+
+	"github.com/digitalocean/godo"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	infrav1 "sigs.k8s.io/cluster-api-provider-digitalocean/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-provider-digitalocean/cloud/scope"
+)
+
+// GetDroplet get a droplet instance
+func (s *Service) GetDroplet(id string) (*godo.Droplet, error) {
+	if id == "" {
+		s.scope.Info("DOMachine does not have an instance id")
+		return nil, nil
+	}
+
+	s.scope.V(2).Info("Looking for instance by id", "instance-id", id)
+	dropletID, err := strconv.Atoi(id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse instance id with id %q", id)
+	}
+
+	droplet, res, err := s.scope.Droplets.Get(s.ctx, dropletID)
+	if err != nil {
+		if res.StatusCode == http.StatusNotFound {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	if droplet != nil {
+		return droplet, nil
+	}
+	return nil, nil
+}
+
+// CreateDroplet create a droplet instance.
+func (s *Service) CreateDroplet(scope *scope.MachineScope) (*godo.Droplet, error) {
+	s.scope.V(2).Info("Creating an instance for a machine")
+
+	bootstrapData, err := base64.StdEncoding.DecodeString(*scope.Machine.Spec.Bootstrap.Data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode bootstrap data")
+	}
+
+	clusterName := infrav1.DOSafeName(s.scope.Name())
+	instanceName := infrav1.DOSafeName(scope.Name())
+	volumeID := scope.InstanceVolumeRef().ResourceID
+
+	image, err := s.GetImage(scope)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed getting image")
+	}
+
+	sshkeys := []godo.DropletCreateSSHKey{}
+	for _, v := range scope.DOMachine.Spec.SSHKeys {
+		keys, err := s.GetSSHKey(v)
+		if err != nil {
+			return nil, err
+		}
+		sshkeys = append(sshkeys, godo.DropletCreateSSHKey{
+			ID:          keys.ID,
+			Fingerprint: keys.Fingerprint,
+		})
+	}
+
+	request := &godo.DropletCreateRequest{
+		Name:    instanceName,
+		Region:  s.scope.Region(),
+		Size:    scope.DOMachine.Spec.Size,
+		SSHKeys: sshkeys,
+		Image: godo.DropletCreateImage{
+			ID: image.ID,
+		},
+		UserData:          string(bootstrapData),
+		PrivateNetworking: true,
+		Volumes: []godo.DropletCreateVolume{
+			{
+				ID: volumeID,
+			},
+		},
```

Comment:
What is the reason for creating/attaching a volume? For what is the volume used?

Reply:
answered here https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/140#discussion_r382921658

Reply:
yeah, i thought it's not useful. i have just updated to remove this one.